modules/library/referencing/src/main/java/org/geotools/referencing/operation/matrix/GeneralMatrix.java

```diff
@@ -873,12 +972,38 @@ public void setSize(int numRows, int numCols) {
         }
     }
 
-    public void sub(GeneralMatrix matrix) {
-        CommonOps.subtract(mat, matrix.mat, mat);
+    @Override
```

Comment:
> The sub public methods lack a javadoc

Darn you are correct, the javadocs were provided by the former superclass.

Adding javadocs to XMatrix, aiming for method compatibility (should help prevent code casting to GeneralMatrix as well).